src/main/java/org/spongepowered/api/command/parameter/Parameter.java

```diff
@@ -713,6 +715,32 @@ static Parameter seq(@NonNull final Iterable<Parameter> parameters) {
         return Parameter.builder(type, vvp.build());
     }
 
+    @SafeVarargs
+    static Function<CommandContext, RegistryHolder> registryFromOptions(final Function<CommandContext, Optional<RegistryHolder>>... functions){
```

Comment:
@dualspiral I still don't know how we want to have this documented, but there's the use case that we already discussed before with the testplugin.

Reply:
I actually need to think about where that should go full stop - I always said here was just a temporary home after all.

Reply:
I've handled this by splitting apart getting the holder (of which multiple can be specified, choosing the first one it comes across) and `RegistryType` to get.

For reference (some of these already existed)

https://github.com/SpongePowered/SpongeAPI/blob/a5b9d9fb5dd5b84a07a8557e9c2348788112b05a/src/main/java/org/spongepowered/api/command/parameter/managed/standard/VariableValueParameters.java#L231-L263

for use with 

https://github.com/SpongePowered/SpongeAPI/blob/a5b9d9fb5dd5b84a07a8557e9c2348788112b05a/src/main/java/org/spongepowered/api/command/parameter/Parameter.java#L750-L761